.travis.yml

```diff
@@ -1,8 +1,24 @@
 language: python
-python:
- - "2.7"
- - "3.6"
-install: pip install -r requirements.txt -r test-requirements.txt
-script:
-    - flake8 --filename="*.py"
-    - nosetests
+before_install:
+  - pip install wheel
+install:
+  - pip install -r requirements.txt -r test-requirements.txt
+cache:
+  directories:
+    - $HOME/.cache/pip
+matrix:
+  include:
+    - name: "Python 2.7 tests"
+      python: "2.7"
+      script: nosetests
+    - name: "Python 2.7 flake8"
+      python: "2.7"
+      install: pip install -r test-requirements.txt
```

Comment:
do we need to duplicate the install in line 4,5 ?

Reply:
yep. Normally all the items in the include would inherit the default install line which is `pip install -r requirements.txt -r test-requirements.txt`

In order to save time, we do not need to install the `requirements.txt` part as this tests as they only run flake so we need to overload the install part on each one.

TBH, with the caching mechanism introduced here, the install part of the whole thing is reduced to ~9 seconds, so it may not make sense to avoid installing the `requirements.txt` file as we do not save that much time during the install (~9s VS ~1,5s)